pkg/evaluator/evaluate.go

```diff
@@ -0,0 +1,153 @@
+package evaluator
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+
+	om "github.com/checkr/openmock"
+	models "github.com/checkr/openmock/swagger_gen/models"
+	"github.com/fatih/structs"
+	"github.com/sirupsen/logrus"
+)
+
+var Evaluate = func(context *models.EvalContext, mock *om.Mock) (response models.MockEvalResponse, err error) {
+	actions_performed := &[]*models.ActionPerformed{}
+
+	// check if the mock's Expect matches the input context (e.g. HTTP path & method)
+	expect_passed := checkChannelCondition(context, mock)
+	if !expect_passed {
+		return models.MockEvalResponse{
+			ExpectPassed:     false,
+			ActionsPerformed: *actions_performed,
+		}, nil
+	}
+
+	// get the OM context that will be used to check condition and perform actions
+	om_context, err := conditionContext(context)
+	if err != nil {
+		logrus.Errorf("Problem setting up om context %v", err)
+		return models.MockEvalResponse{
+			ExpectPassed:     true,
+			ActionsPerformed: *actions_performed,
+		}, err
+	}
+	if om_context != nil {
+		om_context.Values = mock.Values
+	}
+
+	// TODO should we set om_context.currentMock? need to make it not private in OM model if so
```

Comment:
currentMock was introduced for logging, do you need that for logging? I guess it doesn't hurt to make it public and also we can see some logging even it's just for evaluation.

Reply:
Hm, I think in this context you're only sending in one mock at a time, so in terms of logging I don't think it's helpful (you already know the mock).  when I wrote the TODO I was more worried that something we reused in handle_actions might use it, but if it's just for logging that's not really the case.  I'll drop this TODO.